homeassistant/components/light/niko_home_control.py

```diff
@@ -0,0 +1,89 @@
+"""
+Support for Niko Home Control.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/light.niko_home_control/
+"""
+import logging
+import socket
+
+import voluptuous as vol
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, PLATFORM_SCHEMA, Light)
+from homeassistant.const import CONF_HOST
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['niko-home-control==0.1.8']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Niko Home Control light platform."""
+    import nikohomecontrol
+
+    host = config.get(CONF_HOST)
+
+    try:
+        hub = nikohomecontrol.Hub({
+            'ip': host,
+            'port': 8000,
+            'timeout': 20000,
+            'events': True
+        })
+    except socket.error as err:
```

Comment:
Socket error is deprecated in favor of `OSError` in modern versions of Python 3.